ethereum/p2p/src/test/java/tech/pegasys/pantheon/ethereum/p2p/discovery/PeerDiscoveryTimestampsTest.java

```diff
@@ -26,126 +28,107 @@
 import tech.pegasys.pantheon.ethereum.p2p.peers.PeerBlacklist;
 import tech.pegasys.pantheon.ethereum.p2p.permissioning.NodeWhitelistController;
 import tech.pegasys.pantheon.ethereum.permissioning.PermissioningConfiguration;
+import tech.pegasys.pantheon.util.Subscribers;
 
 import java.util.Collections;
+import java.util.List;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicLong;
 
-import io.vertx.core.Vertx;
 import org.junit.Test;
 
-public class PeerDiscoveryTimestampsTest extends AbstractPeerDiscoveryTest {
+public class PeerDiscoveryTimestampsTest {
+  private final PeerDiscoveryTestHelper helper = new PeerDiscoveryTestHelper();
 
   @Test
   public void lastSeenAndFirstDiscoveredTimestampsUpdatedOnMessage() {
     // peer[0] => controller // peer[1] => sender
-    final SECP256K1.KeyPair[] keypairs = PeerDiscoveryTestHelper.generateKeyPairs(2);
-    final DiscoveryPeer[] peers = PeerDiscoveryTestHelper.generateDiscoveryPeers(keypairs);
+    final List<KeyPair> keypairs = PeerDiscoveryTestHelper.generateKeyPairs(2);
+    final List<DiscoveryPeer> peers = helper.createDiscoveryPeers(keypairs);
 
-    final PeerDiscoveryAgent agent = mock(PeerDiscoveryAgent.class);
-    when(agent.getAdvertisedPeer()).thenReturn(peers[0]);
+    final MockPeerDiscoveryAgent agent = mock(MockPeerDiscoveryAgent.class);
+    when(agent.getAdvertisedPeer()).thenReturn(peers.get(0));
+    DiscoveryPeer self = peers.get(0);
```

Comment:
Again, `self` as a variable may cause python confusion with `this`, an alternate name would improve readability.